integrations/jta/jdbc/src/main/java/io/helidon/integrations/jta/jdbc/JtaConnection.java

```diff
@@ -1124,27 +1124,18 @@ private Connection connectionFunction(Xid xid) {
     private void transactionCompleted(int commitedOrRolledBack) {
         this.enlistment = null; // volatile write
         try {
-            if (!super.isClosed()) {
-                // Did someone call close() while we were enlisted?  That's permitted by section 4.2 of the JTA
-                // specification, but it shouldn't actually close the connection because the prepare/commit cycle
-                // wouldn't have started.
-                boolean closeWasPending = this.isClosePending(); // volatile read
-
-                // If they did, then we must be non-closeable. If they didn't, we could be either closeable or not
-                // closeable.
-                assert !closeWasPending || !this.isCloseable();
-
-                // Now the global transaction is over, so blindly set our closeable status to true. (It may already be
-                // true.)  This resets the closePending status, per spec.
-
-                this.setCloseable(true);
-                assert this.isCloseable();
+            boolean closeWasPending = this.isClosePending();
```

Comment:
An informational question... (Maybe we discussed this in an earlier round of reviews on related PRs and I've forgotten. Also, this is not directly related to this latest set of changes.) 

The group of flags (`isCloseable`, `isClosePending`, `isClosed`) suggest that there's a state model lurking here. I wonder if being explicit about the states and the transitions among them might clarify things or would, in fact, actually muddy the waters.

Reply:
Yes. The trick is that the state model has to play nice with the state model of `java.sql.Connection`, which exposes `close()` and `isClosed()`.

(I also can't change the signatures of the public classes, like `ConditionallyCloseableConnection`, that already exist.)

The state model is described in javadoc; see https://helidon.io/docs/v3/apidocs/io.helidon.integrations.jdbc/io/helidon/integrations/jdbc/ConditionallyCloseableConnection.html.

In general, if the `closeable` property is `false`, then attempts to close the connection will record that a close request happened.  `isClosed`, in order to honor the `java.sql.Connection` contract, will always return `true` once `close()` has been invoked. Clear as mud? The spirit of this contract, while not reified anywhere other than these Helidon classes, is pervasive throughout the JDBC specification.